nixio/pycore/h5dataset.py

```diff
@@ -43,6 +43,8 @@ def write_data(self, data, count=None, offset=None):
             self.dataset[:] = data
 
     def read_data(self, data, count=None, offset=None):
+        if 0 in self.dataset.shape or len(self.dataset.shape) == 0:
+            return
```

Comment:
I need to correct my claim about matlab, though I am kind-of sure, I have seen this at some point. Anyway, the current state is that matlab does not have dimensions that have a zero size. Similar to numpy zeros(2,0,3) gives an empty matrix with the given shape and the size 0. 
HDF does not seem to have a problem with storing this dimension information and hdfview just opens the file. It seems, though that nix does not like it either. 

Regarding the desired behaviour in nixpy, I do not have a strong opinion. Maybe act as numpy does and return an empty variable of the given shape ```np.zeros(shape)```  instead of returning ```None```.